### DISCUSSION TITLE
Way to Cluster Detections by Similarity?

### DISCUSSION THREAD

**Discussion by quitmeyer** [ultralytics/ultralytics#14728] [General]
Hi! I am building a insect sensor and have already made a detector with Yolo that has been working great to detect and crop images of individual insects.

So it outputs a nice folder of cropped insect images. But now, We need to try to organize insects that might seem similar (or are photos of the same insect at slightly different times in the night). This is because we are then giving the data to some scientists who want to quickly go through all the images and give them labels, and it's WAYYY easier if they can look at a cluster of 100 images that all look pretty much the same and give them all the same label.

I did a VERY BASIC clustering by just organizing the images by image width so you can see even this makes things easier, but some images still get mixed with each other.
![image](https://github.com/user-attachments/assets/3e09b5bc-8b71-45fd-9187-463142de2baf)

Is there a thing like Yolo that is good for just looking at a folder of images and clustering them?

**Comment by pderrenger** [comment]
@quitmeyer hi! It's great to hear that YOLO is working well for your insect detection project. For clustering similar images, you might want to explore feature extraction techniques using pre-trained models. One approach is to use a model like YOLOv8 for feature extraction and then apply clustering algorithms such as K-means or DBSCAN on these features.

You can extract features from the images using a pre-trained model and then cluster these features to group similar images together. This should help in organizing your insect images more effectively for the scientists to label.

If you encounter any issues or need further assistance, feel free to ask.
